File: src/main/scala/info/crlog/higgs/protocol/MessageListener.scala
package info.crlog.higgs.protocol

import org.jboss.netty.channel.{ChannelStateEvent, ChannelHandlerContext}

/**
 * @author Courtney Robinson <courtney@crlog.info> @ 01/02/12
 */

trait MessageListener {
  def onMessage(msg: Message)
  def onConnected(ctx: ChannelHandlerContext, e: ChannelStateEvent): Unit ={

  }
}

File: src/main/java/info/crlog/higgs/messaging/MessageListener.java
package info.crlog.higgs.messaging;

/**
 * While higgs agents such as Broadcaster or radio uses HiggsEventListener Use
 * classes should only be interested in the message received event.
 *
 * @author Courtney Robinson <courtney@crlog.info>
 */
public interface MessageListener {

    /**
     * Invoked when a Message is received
     *
     * @param <M extends Message>
     * @param message
     */
    public <M extends Message> void onMessage(M message);
}
